keps/sig-storage/5040-remove-gitrepo-driver/README.md

```diff
@@ -0,0 +1,1047 @@
+* gitRepo has low usage (based on limited data)
+* there exists precedent for removing volume drivers which were unmaintained like [glusterfs](https://kubernetes.io/docs/concepts/storage/volumes/#glusterfs)
+
+Given this, we think kubernetes should EOL this feature with urgency.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Remove a vector for remote code execution through an ancient and unmaintained
+volume plugin.
+
+SIG Storage was actually very close to doing this and had even added a notice of
+removal in the release notes for v1.32. But due to concerns raised by SIG Architecture
+they rolled back the notice. SIG Storage provided [this rationale](https://github.com/kubernetes/kubernetes/issues/125983#issuecomment-2522201283) for considering removal.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Remove in-tree gitRepo volume support.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+- Remove gitRepo volumes from the Kubernetes API.
+
+## Proposal
+
+Kubernetes drops support for gitRepo volume types by removing the in-tree driver
+that actually mounts the volume in the pod. In this case we will still keep the
+gitRepo volume in the API. Pods with gitRepo volumes will be admitted by 
+kube-apiserver however, kubelet will fail to run them with an appropriate 
+error message.
+
+
+This removal is similar to removal of support for
+[glusterfs](https://kubernetes.io/docs/concepts/storage/volumes/#glusterfs)
+which was deprecated in v1.25 and removed in v1.26.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+The biggest risk is in breaking users. Despite being deprecated for years, there
+may be some users who are still using it and their workloads would be broken 
+when their clusters are upgraded. 
+
+We plan to mitigate the risk by:
+1. Allowing users to opt-in to re-enabling the driver for 3 versions to give
+  them enough time to fix  workloads
+2. Announcing the change via release notes, updates to the gitRepo documentation
+  and publishing kubernetes release blog for this feature so that customers are
+  aware and can take action before they upgrade
```

Comment:
Should probably include some language about communicating alternatives?

Reply:
Good idea. Added.